number_theory/modular_unittest.cpp

```diff
@@ -0,0 +1,41 @@
+#include <gtest/gtest.h>
+
+#include "number_theory/modular.h"
+
+namespace tql {
+
+TEST(ModularTest, Basic) {
+  using Mod10 = Modular<int, 10>;
+
+  // test constructors
+  Mod10 a = 123;
+  EXPECT_EQ(a.get(), 3);
+  Mod10 b = a;
+  EXPECT_EQ(b.get(), 3);
+  Mod10 c;
```

Comment:
I think we should also check the default constructor's value like `EXPECT_EQ(c.get(), 0)`.